src/ucp/core/ucp_worker.c

```diff
@@ -1419,33 +1487,35 @@ static void ucp_worker_init_guess_atomics(ucp_worker_h worker)
     }
 
     if (accumulated_flags & UCT_IFACE_FLAG_ATOMIC_DEVICE) {
-        ucp_worker_init_device_atomics(worker);
-    } else {
-        ucp_worker_init_cpu_atomics(worker);
+        return ucp_worker_init_device_atomics(worker);
     }
+
+    ucp_worker_init_cpu_atomics(worker);
+    return UCS_OK;
 }
 
-static void ucp_worker_init_atomic_tls(ucp_worker_h worker)
+static ucs_status_t ucp_worker_init_atomic_tls(ucp_worker_h worker)
 {
     ucp_context_h context = worker->context;
 
     worker->atomic_tls = 0;
+    worker->atomic_ep  = NULL;
 
-    if (context->config.features & UCP_FEATURE_AMO) {
-        switch(context->config.ext.atomic_mode) {
-        case UCP_ATOMIC_MODE_CPU:
-            ucp_worker_init_cpu_atomics(worker);
-            break;
-        case UCP_ATOMIC_MODE_DEVICE:
-            ucp_worker_init_device_atomics(worker);
-            break;
-        case UCP_ATOMIC_MODE_GUESS:
-            ucp_worker_init_guess_atomics(worker);
-            break;
-        default:
-            ucs_fatal("unsupported atomic mode: %d",
-                      context->config.ext.atomic_mode);
-        }
+    if (!(context->config.features & UCP_FEATURE_AMO)) {
```

Comment:
👍 this is example of a case where "else" -> "if (!cond) return" DOES make sense..